### PR TITLE
feat: update otel versions for prometheus to 0.26

### PR DESCRIPTION
**Pull request by ouertani**
Fixes #
Design discussion issue (if applicable) #
align `opentelemetry-prometheus` to `0.26.0` version 

## Changes

Please provide a brief description of the changes here.

- update create dependency to 0.26 
- remove  `AggregationSelector`  as per https://github.com/open-telemetry/opentelemetry-rust/pull/2085 
- remove `MetricProducer`  as per https://github.com/open-telemetry/opentelemetry-rust/pull/2105
- use KeyValue as per https://github.com/open-telemetry/opentelemetry-rust/pull/2091

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by cijothomas — APPROVED** [review]
Thanks. Please update changelog.md too.

Also, see https://github.com/open-telemetry/opentelemetry-rust/pull/1957#pullrequestreview-2196978183
No change in Prometheus status - it may not work in next release! if it works, they good, but no guarantees.

**Comment by lalitb** [comment]
@ouertani Do you want to look into the CI tests for Prometheus which are failing right now ?

**Comment by ouertani** [comment]
> @ouertani Do you want to look into the CI tests for Prometheus which are failing right now ?

Yes, I will try to fix

**Comment by ouertani** [comment]
@lalitb this pr is required to fix the tests https://github.com/open-telemetry/opentelemetry-rust/pull/2224 seems introduced with this pr https://github.com/open-telemetry/opentelemetry-rust/pull/2018

**Comment by cijothomas** [comment]
> @lalitb this pr is required to fix the tests #2224 seems introduced with this pr #2018

See https://github.com/open-telemetry/opentelemetry-rust/pull/2224#pullrequestreview-2382971228
To unblock this PR, you may Ignore the test about duplicate attributes, and re-enable once we fix the issue in the sdk.

**Comment by ouertani** [comment]
> fix: ignore prometheus_exporter_integration until fixed with sdk

[fix: ignore prometheus_exporter_integration until fixed with sdk](https://github.com/open-telemetry/opentelemetry-rust/pull/2183/commits/427d0ff2d0412589851219f2c5059b2460c16f2c)

File: opentelemetry-prometheus/tests/integration_test.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@ use opentelemetry_sdk::Resource;
 use opentelemetry_semantic_conventions::resource::{SERVICE_NAME, TELEMETRY_SDK_VERSION};
 use prometheus::{Encoder, TextEncoder};
 
+#[ignore]
```
Comment (cijothomas) [inline opentelemetry-prometheus/tests/integration_test.rs:18]: please point to the issue or the PRs trying to fix this for easy referenc.e

Comment (ouertani) [reply opentelemetry-prometheus/tests/integration_test.rs]: I add the pr link

**Review by lalitb — APPROVED** [review]

**Pull request merged**